shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -3482,5 +3482,204 @@ TEST_F(EmbedderTest, ArcEndCapsAreDrawnCorrectly) {
   ASSERT_TRUE(ImageMatchesFixture("arc_end_caps.png", scene_image));
 }
 
+TEST_F(EmbedderTest, ObjectsCanBePostedViaPorts) {
+  auto& context = GetEmbedderContext();
+  EmbedderConfigBuilder builder(context);
+  builder.SetOpenGLRendererConfig(SkISize::Make(800, 1024));
+  builder.SetDartEntrypoint("objects_can_be_posted");
+
+  // Synchronously acquire the send port from the Dart end. We will be using
+  // this to send message. The Dart end will just echo those messages back to us
+  // for inspection.
+  FlutterEngineDartPort port = 0;
+  fml::AutoResetWaitableEvent event;
+  context.AddNativeCallback("SignalNativeCount",
+                            CREATE_NATIVE_ENTRY([&](Dart_NativeArguments args) {
+                              port = tonic::DartConverter<int64_t>::FromDart(
+                                  Dart_GetNativeArgument(args, 0));
+                              event.Signal();
+                            }));
+  auto engine = builder.LaunchEngine();
+  ASSERT_TRUE(engine.is_valid());
+  event.Wait();
+  ASSERT_NE(port, 0);
+
+  using Trampoline = std::function<void(Dart_Handle message)>;
+  Trampoline trampoline;
+
```

Comment:
Is this a common pattern, registering native functions after the isolate is already running? I'll have to keep in mind that some embedders are relying on native function resolution being lazy.

Reply:
I don't see how this is of consequence to the Dart VM. The Dart embedder just sets up a single native resolver via `Dart_SetNativeResolver` before the isolate has been launched. This fixture just adds another entry to the map of procs that gets queries in the callback.

Reply:
Oh. Did you mean that the Dart_NativeFunction that gets returned by the Dart_NativeEntryResolver is somehow cached? Then yeah, the test harness does depend on not having this cached by the VM. But if you change this behavior in the future, actual applications will not be affected as they don't rely on lazy resolution. If/when that happens, I can modify the test harness as well.

Reply:
> Did you mean that the Dart_NativeFunction that gets returned by the Dart_NativeEntryResolver is somehow cached?

Scratch that. Even the Dart code in the test harness takes care not the invoke native entrypoints before their handlers have been registered.

Reply:
It's relying on the native resolver not being invoked with each native until the it is used the first time. A counterfactual implementation of the VM could compile all functions and resolve native functions at isolate start.

Reply:
Ok, I think the tests are safe in that regard. While the hypothetical VM you described would definitely break the tests, actual applications should be fine as they don't run afoul of the behavior you described. Also, feel free to change this behavior if necessary. Reworking the existing tests should not be an issue. In the meantime, I will think of migrating away from this pattern anyway as this is admittedly relying on undocumented behavior.